libparsec/crates/platform_device_loader/src/native/mod.rs

```diff
@@ -322,17 +327,52 @@ fn save_device_file(key_file: &PathBuf, file_content: &DeviceFile) -> Result<(),
     Ok(())
 }
 
+fn generate_keyring_user(
+    keyring_user_path: &PathBuf,
+) -> Result<(SecretKey, String), SaveDeviceError> {
+    // Generate a keyring user
+    let keyring_user = Uuid::new_v4().to_string();
+
+    // Generate a key
+    let (passphrase, key) = SecretKey::generate_recovery_passphrase();
+
+    let entry = KeyringEntry::new(KEYRING_SERVICE, &keyring_user)?;
+
+    // Add the key to the keyring
+    entry.set_password(&passphrase)?;
+
+    // Save the keyring user to the config file
+    save_content(keyring_user_path, keyring_user.as_bytes())?;
+
+    Ok((key, keyring_user))
+}
+
 pub async fn save_device(
     access: &DeviceAccessStrategy,
     device: &LocalDevice,
 ) -> Result<(), SaveDeviceError> {
     match access {
         DeviceAccessStrategy::Keyring { key_file } => {
-            let entry = KeyringEntry::new(KEYRING_SERVICE, &device.slughash())?;
-
-            let (passphrase, key) = SecretKey::generate_recovery_passphrase();
-
-            entry.set_password(&passphrase)?;
+            let keyring_user_path = get_default_config_dir().join("keyring_user.txt");
+
+            let (key, keyring_user) = if keyring_user_path.exists() {
+                let keyring_user = std::fs::read_to_string(&keyring_user_path).map_err(|_| {
+                    SaveDeviceError::Internal(anyhow::anyhow!("Cannot read keyring user file"))
```

Comment:
I would treat this error as for non-existing file, since there's a slight chance we might be able to overwrite this file.

Reply:
This error can't happen since I check if the file exists before

Reply:
The error can append ;)

https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

Better directly call `read_to_string` and handle the error in case the file doesn't exist

Reply:
Oh it's even better :+1:

Reply:
Done